eo-maven-plugin/src/main/java/org/eolang/maven/VerifyMojo.java

```diff
@@ -23,28 +23,45 @@
  */
 package org.eolang.maven;
 
+import com.jcabi.log.Logger;
+import com.yegor256.xsline.TrDefault;
 import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.Parameter;
+import org.cactoos.iterable.Filtered;
+import org.cactoos.map.MapEntry;
+import org.cactoos.map.MapOf;
+import org.eolang.maven.optimization.OptTrain;
+import org.eolang.maven.optimization.Optimization;
+import org.eolang.maven.tojos.ForeignTojo;
 
 /**
  * Mojo that checks errors and warnings after "assemble" phase.
  *
  * @since 0.31.0
- * @todo #1708:30min Implement VerifyMojo. VerifyMojo should check all errors
- *  and critical errors in xmir after {@link AssembleMojo} is finished. Also if
- *  {@code failOnWarning} flag is set to true - mojo should check warnings. When
- *  mojo is implemented - need to remove "failOnError" flag from
- *  {@link OptimizeMojo} and put "verify" step right after "assemble" in all
- *  pom.xml files
  */
 @Mojo(
     name = "verify",
     defaultPhase = LifecyclePhase.PROCESS_SOURCES,
     threadSafe = true
 )
 public final class VerifyMojo extends SafeMojo {
+
+    /**
+     * Whether we should fail on error.
+     *
+     * @checkstyle MemberNameCheck (7 lines)
+     * @since 0.34.0
+     */
+    @SuppressWarnings("PMD.ImmutableField")
+    @Parameter(
+        property = "eo.failOnError",
+        defaultValue = "true")
+    private boolean failOnError = true;
```

Comment:
@levBagryansky the main idea of the VerifyMojo was to get rid of `failOnError` flag. More details [here](https://github.com/objectionary/eo/issues/1708). Long story short - if there's at least one ERROR - throw an exception. If there's at least on WARNING - throw an exception if `failOnWarning` is true

Reply:
@maxonfjvipon done